resolwe/flow/models/entity.py

```diff
@@ -58,27 +59,59 @@ class Meta(BaseCollection.Meta):
             ("owner_relation", "Is owner of the relation"),
         )
 
+    UNIT_HOUR = 'hr'
+    UNIT_CHOICES = (
+        (UNIT_HOUR, 'Hour'),
+    )
+
+    def validate_unique(self, exclude=None):
+        """Check unique constraints on the model.
+
+        Separate check for the combination of ``collection`` and ``category``
+        is required as ``category`` is optional and thus
+        ``Meta.unique_together`` cannot be used.
+        """
+        super().validate_unique(exclude)
+
+        duplicate_exists = Relation.objects.exclude(id=self.id).filter(
+            collection_c=self.collection, category=self.category, category__isnull=False
+        ).exists()
+
+        if duplicate_exists:
+            raise ValidationError("Combination of 'collection' and 'category' must be unique.")
+
     #: type of the relation
     type = models.ForeignKey('RelationType', on_delete=models.PROTECT)
 
     #: collection in which relation is
     collection = models.ForeignKey('Collection', null=True, on_delete=models.SET_NULL)
```

Comment:
I should probably delete the relations if I delete the collection. Otherwise it may break the `validate_unique` constraint. This would cause deleting the collection to fail... Say you delete a collection with Treatment category. Then you delete another collection with Treatment category. This will break the unique constraint because you can not have two Treatment relations with NULL collection.